nrf52-hal-common/src/uarte.rs

```diff
@@ -163,6 +164,130 @@ impl<T> Uarte<T> where T: UarteExt {
         Ok(())
     }
 
+    /// Read via UARTE
+    ///
+    /// This method fills all bytes in `rx_buffer`, and blocks
+    /// until the buffer is full.
+    ///
+    /// The buffer must have a length of at most 255 bytes
+    pub fn read(&mut self,
+        rx_buffer  : &mut [u8],
+    )
+        -> Result<(), Error>
+    {
+        self.start_read(rx_buffer)?;
+
+        // Wait for transmission to end
+        while self.0.events_endrx.read().bits() == 0 {}
+
+        self.finalize_read();
+
+        if self.0.rxd.amount.read().bits() != rx_buffer.len() as u32 {
+            return Err(Error::Receive);
+        }
+
+        Ok(())
+    }
+
+    /// Read via UARTE
+    ///
+    /// This method fills all bytes in `rx_buffer`, and blocks
+    /// until the buffer is full.
+    ///
+    /// The buffer must have a length of at most 255 bytes
+    pub fn read_timeout<I>(
+        &mut self,
+        rx_buffer: &mut [u8],
+        timer: &mut Timer<I>,
+        cycles: u32
+    ) -> Result<(), Error> where I: TimerExt
+    {
+        // Start the read
+        self.start_read(rx_buffer)?;
+
+        // Start the timeout timer
+        timer.start(cycles);
+
+        // Wait for transmission to end
+        let mut event_complete = false;
+        let mut timeout_occured = false;
+
+        loop {
```

Comment:
I think this is fine, but we should probably document that this method involves busy waiting.